### DISCUSSION TITLE
using this library from .NET Framework

### DISCUSSION THREAD

**Discussion by EsiKhoob** [libass/libass#628] [Q&A] [answered]
Hi
I am a .NET developer. As you know, .NET is a Framework from Microsoft, which works with C# language.
I want to hardcode srt files into different video file format. can I do this with libass? 
My another basic question is: Can I call your library from with in my .NET code? I think I need a wrapper library. Is there any? I am aware of ffmpeg wrappers for .NET, but I want subtitle hardcoding capability only.
Thanks in advance.

**Comment by TheOneric** [comment, marked as answer]
> Can I call your library from with in my .NET code?

If C# can interface with standard C libraries, then supposedly yes.

> I think I need a wrapper library. Is there any?

I do not know.

> I want to hardcode srt files into different video file format. can I do this with libass?

Short version: Yes, but not directly.

Long version: libass is a renderer for ASS not SRT. In order to render SRT you will first need to convert SRT data to ASS. In practice many SRT files out there actually *do* expect to be rendered in such a way by an ASS renderer and contain active ASS sequences like e.g. `{\fade(800,800)}`. Therefore using a conversion + ASS renderer might actually be the only way to render some files as intended. To that end, you must not escape active ASS sequences when doing the conversion. However, SRT files also sometimes contain HTML tags like `<i>...</i>`; those have no meaning in ASS and you must translate them to an ASS equivalent during conversion.

ffmpeg can already do this conversion for you, but if you absolutely want to avoid ffmpeg, you'll have to roll your own conversion logic and be(come) familiar with both ASS and SRT as they are used in real world files.
If it fits into your project, you might want to consider using a barebones ffmpeg afterall to avoid having to create and maintain your own converted and possibly C#-wrapper.
